tests/api/reports-import.php

```diff
@@ -0,0 +1,154 @@
+ */
+
+/**
+ * Reports Import REST API Test Class
+ *
+ * @package WooCommerce\Tests\API
+ */
+class WC_Tests_API_Reports_Import extends WC_REST_Unit_Test_Case {
+	/**
+	 * Endpoint.
+	 *
+	 * @var string
+	 */
+	protected $endpoint = '/wc/v4/reports/import';
+
+	/**
+	 * Setup test reports products data.
+	 */
+	public function setUp() {
+		parent::setUp();
+
+		$this->user = $this->factory->user->create(
+			array(
+				'role' => 'administrator',
+			)
+		);
+	}
+
+	/**
+	 * Test route registration.
+	 */
+	public function test_register_routes() {
+		$routes = $this->server->get_routes();
+
+		$this->assertArrayHasKey( $this->endpoint, $routes );
+	}
+
+	/**
+	 * Asserts the report item schema is correct.
+	 *
+	 * @param array $schema Item to check schema.
+	 */
+	public function assert_report_item_schema( $schema ) {
+		$this->assertArrayHasKey( 'status', $schema );
+		$this->assertArrayHasKey( 'message', $schema );
+	}
+
+	/**
+	 * Test reports schema.
+	 */
+	public function test_reports_schema() {
+		wp_set_current_user( $this->user );
+
+		$request    = new WP_REST_Request( 'OPTIONS', $this->endpoint );
+		$response   = $this->server->dispatch( $request );
+		$data       = $response->get_data();
+		$properties = $data['schema']['properties'];
+
+		$this->assertCount( 2, $properties );
+		$this->assert_report_item_schema( $properties );
+	}
+
+	/**
+	 * Test getting reports without valid permissions.
+	 */
+	public function test_get_reports_without_permission() {
+		wp_set_current_user( 0 );
+		$response = $this->server->dispatch( new WP_REST_Request( 'POST', $this->endpoint ) );
+		$this->assertEquals( 401, $response->get_status() );
+	}
+
+	/**
+	 * Test the import paramaters.
+	 */
+	public function test_import_params() {
+		global $wpdb;
+		wp_set_current_user( $this->user );
+
+		// Populate all of the data.
+		$product = new WC_Product_Simple();
+		$product->set_name( 'Test Product' );
+		$product->set_regular_price( 25 );
+		$product->save();
+
+		$order_1 = WC_Helper_Order::create_order( 1, $product );
+		$order_1->set_status( 'completed' );
+		$order_1->set_date_created( time() - ( 3 * DAY_IN_SECONDS ) );
+		$order_1->save();
+		$order_2 = WC_Helper_Order::create_order( 1, $product );
+		$order_2->set_total( 100 );
+		$order_2->set_status( 'completed' );
+		$order_2->save();
+
+		// Delete order stats so we can test import API.
+		$wpdb->query( "DELETE FROM {$wpdb->posts} WHERE post_type = 'scheduled-action'" );
+		$wpdb->query( "DELETE FROM {$wpdb->prefix}wc_order_stats" );
+
+		// Use the days param to only process orders in the last day.
+		$request = new WP_REST_Request( 'POST', $this->endpoint );
+		$request->set_query_params( array( 'days' => '1' ) );
+		$response = $this->server->dispatch( $request );
+		$report   = $response->get_data();
+
+		$this->assertEquals( 200, $response->get_status() );
+		$this->assertEquals( 'success', $report['status'] );
+
+		// Run pending twice to process batch and order.
```

Comment:
Really like the test case here - given that in Jeff's prior review there seemed to be some issues around customer creation logic, do you think we should add some test coverage around that too? Or do we have unit test coverage elsewhere that should cover that?